test/e2e_node/split_disk_test.go

```diff
@@ -0,0 +1,309 @@
+	evictionapi "k8s.io/kubernetes/pkg/kubelet/eviction/api"
+	kubeletmetrics "k8s.io/kubernetes/pkg/kubelet/metrics"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	"k8s.io/kubernetes/test/e2e/nodefeature"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+var _ = SIGDescribe("SplitDisk", nodefeature.SplitDisk, func() {
+	f := framework.NewDefaultFramework("split-disk-test")
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+	pressureTimeout := 10 * time.Minute
+	expectedNodeCondition := v1.NodeDiskPressure
+
+	ginkgo.BeforeEach(func(ctx context.Context) {
+		e2eskipper.SkipUnlessFeatureGateEnabled(features.KubeletSeparateDiskGC)
+		if !hasSplitFileSystem(ctx) {
+			ginkgo.Skip("it doesn't have split filesystem")
+		}
+	})
+
+	f.It("should display different stats for imagefs and containerfs", func(ctx context.Context) {
+		summary := eventuallyGetSummary(ctx)
+		gomega.Expect(summary.Node.Fs.AvailableBytes).ToNot(gomega.Equal(summary.Node.Runtime.ImageFs.AvailableBytes))
+		gomega.Expect(summary.Node.Fs.CapacityBytes).ToNot(gomega.Equal(summary.Node.Runtime.ImageFs.CapacityBytes))
+		// Node.Fs represents rootfs which /var/lib/kubelet is on.
+		// Since the graphroot is left by default in storage.conf, and it is the same filesystem as rootfs,
+		// Node.Fs should be the same as Runtime.ContainerFs.
+		gomega.Expect(summary.Node.Fs.AvailableBytes).To(gomega.Equal(summary.Node.Runtime.ContainerFs.AvailableBytes))
+		gomega.Expect(summary.Node.Fs.CapacityBytes).To(gomega.Equal(summary.Node.Runtime.ContainerFs.CapacityBytes))
+	})
+
+	f.Context("when there is disk pressure", framework.WithSlow(), framework.WithSerial(), framework.WithDisruptive(), func() {
+		f.Context("on imageFs", func() {
+			tempSetCurrentKubeletConfig(f, func(ctx context.Context, initialConfig *kubeletconfig.KubeletConfiguration) {
+				initialConfig.EvictionHard = map[string]string{string(evictionapi.SignalNodeFsAvailable): "30%", string(evictionapi.SignalContainerFsAvailable): "30%", string(evictionapi.SignalImageFsAvailable): "30%"}
+				initialConfig.EvictionMinimumReclaim = map[string]string{}
+				ginkgo.By(fmt.Sprintf("EvictionHard %s", initialConfig.EvictionHard))
+			})
+
+			runImageFsPressureTest(f, pressureTimeout, expectedNodeCondition, logDiskMetrics, []podEvictSpec{
+				{
+					evictionPriority: 1,
+					pod:              innocentPod(),
+				},
+			})
+		})
+
+		f.Context("on containerFs", func() {
+			expectedStarvedResource := v1.ResourceEphemeralStorage
+			diskTestInMb := 5000
+
+			tempSetCurrentKubeletConfig(f, func(ctx context.Context, initialConfig *kubeletconfig.KubeletConfiguration) {
+				initialConfig.EvictionHard = map[string]string{string(evictionapi.SignalNodeFsAvailable): "30%", string(evictionapi.SignalImageFsAvailable): "30%"}
```

Comment:
Ditto, as the above.

```suggestion
initialConfig.EvictionHard = map[string]string{
        string(evictionapi.SignalNodeFsAvailable): "30%",
        string(evictionapi.SignalImageFsAvailable): "30%",
}
```